src/main/java/io/supertokens/totp/Totp.java

```diff
@@ -0,0 +1,197 @@
+package io.supertokens.totp;
+
+import io.supertokens.Main;
+import io.supertokens.pluginInterface.exceptions.StorageQueryException;
+import io.supertokens.pluginInterface.totp.TOTPDevice;
+import io.supertokens.pluginInterface.totp.TOTPUsedCode;
+import io.supertokens.pluginInterface.totp.exception.DeviceAlreadyExistsException;
+import io.supertokens.pluginInterface.totp.exception.TotpNotEnabledException;
+import io.supertokens.pluginInterface.totp.exception.UnknownDeviceException;
+import io.supertokens.pluginInterface.totp.sqlStorage.TOTPSQLStorage;
+import io.supertokens.storageLayer.StorageLayer;
+import io.supertokens.totp.exceptions.InvalidTotpException;
+import io.supertokens.totp.exceptions.LimitReachedException;
+
+public class Totp {
+
+    public static String generateSecret() {
+        return "XXXX";
+    }
+
+    public static boolean checkCode(TOTPDevice device, String code) {
```

Comment:
```suggestion
    private static boolean checkCode(TOTPDevice device, String code) {
```

Reply:
For testing, you can make a public static function which calls this one, and annotate it with `@TestOnly`

Reply:
Done..